components/dialog/index.css

```diff
@@ -34,337 +34,321 @@ governing permissions and limitations under the License.
-  min-inline-size: var(--spectrum-dialog-min-width);
-
-  padding: var(--spectrum-dialog-padding);
+  width: fit-content;
+  min-width: var(--spectrum-dialog-min-width);
+  max-width: 100%;
 
-  border-radius: var(--spectrum-dialog-border-radius);
-  outline: none; /* Firefox shows outline */
+  max-height: inherit;
 
-  /* Exit animations */
-  transition: opacity var(--spectrum-dialog-exit-animation-duration) var(--spectrum-global-animation-ease-in) var(--spectrum-dialog-exit-animation-delay),
-              visibility 0ms linear calc(var(--spectrum-dialog-exit-animation-delay) + var(--spectrum-dialog-exit-animation-duration)),
-              transform 0ms linear calc(var(--spectrum-dialog-exit-animation-delay) + var(--spectrum-dialog-exit-animation-duration));
-
-  &.is-open {
-    @inherit: %spectrum-overlay--open;
-
-    /* Entry animations */
-    transition: transform var(--spectrum-dialog-entry-animation-duration) var(--spectrum-global-animation-ease-out) var(--spectrum-dialog-entry-animation-delay),
-                opacity var(--spectrum-dialog-entry-animation-duration) var(--spectrum-global-animation-ease-out) var(--spectrum-dialog-entry-animation-delay);
-
-    transform: translate(-50%, calc(-50% + var(--spectrum-dialog-vertical-offset)));
-  }
-
-  .spectrum-Dialog-closeButton {
-    display: none;
-  }
+  outline: none;
 }
 
 .spectrum-Dialog--small {
-  inline-size: var(--spectrum-dialog-small-width);
+  width: var(--spectrum-dialog-small-width);
 }
 
 .spectrum-Dialog--medium {
-  inline-size: var(--spectrum-dialog-medium-width);
+  width: var(--spectrum-dialog-medium-width);
 }
 
 .spectrum-Dialog--large {
-  inline-size: var(--spectrum-dialog-large-width);
+  width: var(--spectrum-dialog-large-width);
 }
 
 .spectrum-Dialog-hero {
-  block-size: var(--spectrum-dialog-hero-height);
-
-  margin-inline-start: calc(-1 * var(--spectrum-dialog-padding));
-  margin-inline-end: calc(-1 * var(--spectrum-dialog-padding));
-  margin-block-start: calc(-1 * var(--spectrum-dialog-padding));
-  margin-block-end: var(--spectrum-dialog-padding);
+  grid-area: hero;
+  height: var(--spectrum-dialog-hero-height);
 
-  border-start-start-radius: var(--spectrum-dialog-border-radius);
-  border-start-end-radius: var(--spectrum-dialog-border-radius);
+  border-top-left-radius: var(--spectrum-dialog-border-radius);
+  border-top-right-radius: var(--spectrum-dialog-border-radius);
 
   background-size: cover;
   background-position: center center;
+  overflow: hidden;
 }
 
-.spectrum-Dialog-header {
-  display: flex;
-  /* Without this, buttons will be stretched */
-  align-items: center;
+.spectrum-Dialog .spectrum-Dialog-grid {
+  display: grid;
+  grid-template-columns: var(--spectrum-dialog-padding) auto 1fr auto minmax(0, auto) var(--spectrum-dialog-padding);
+  grid-template-rows: auto var(--spectrum-dialog-padding) auto auto 1fr auto var(--spectrum-dialog-padding);
+  grid-template-areas:
+    "hero hero    hero    hero        hero        hero"
```

Comment:
nice. I used http://asciiflow.com on the past but this feels kind of self-explaining.

![Screen Shot 2020-05-29 at 17 29 34](https://user-images.githubusercontent.com/52184321/83314990-02c6ab80-a1d2-11ea-9c8b-05181658cc65.png)